webapp/views/App/views/Data/MapView/SamplingPointDataLayer/SamplingPointDataLayer.js

```diff
@@ -11,9 +11,34 @@ const markerRadius = 10
 export const SamplingPointDataLayer = (props) => {
   const { markersColor } = props
 
-  const { clusters, clusterExpansionZoomExtractor, clusterIconCreator, overlayName, totalPoints } =
+  const { clusters, clusterExpansionZoomExtractor, clusterIconCreator, overlayName, totalPoints, items } =
     useSamplingPointDataLayer(props)
 
+  // Have a Reference to points for opening popups automatically
+  let markerRefs = useRef([])
+
+  const openPopupOfUuid = (uuid) => {
+    markerRefs.current[uuid].openPopup()
+  }
+
+  const getPointIndex = (uuid) => {
+    let index
+    items.forEach((p, i) => {
+      if (p.uuid == uuid) index = i
+      return
+    })
```

Comment:
this could be done with a simpler findIndex on items, something like: return items.findIndex(item => item.uuid === uuid)

Reply:
Oh yeah, that is correct. Under the hood it will work the same way but is a bit more readable 👍